src/linear_algebra/projective_space/basic.lean

```diff
@@ -0,0 +1,232 @@
+/-
+Copyright (c) 2022 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import linear_algebra.finite_dimensional
+
+/-!
+
+# Projective Spaces
+
+This file contains the definition of the projectivization of a vector space over a field,
+as well as the bijection between said projectivization and the collection of all one
+dimensional subspaces of the vector space.
+
+## Constructing terms of `projectivization K V`.
+We have three ways to construct terms of `projectivization K V`:
+- `projectivization.mk K v hv` where `v : V` and `hv : v ≠ 0`.
+- `projectivization.mk' K v` where `v : { w : V // w ≠ 0 }`.
+- `projectivization.mk'' H h` where `H : submodule K V` and `h : finrank H = 1`.
+
+## Other definitions
+- For `v : projectivization K V`, `v.submodule` gives the corresponding submodule of `V`.
+- `projectivization.equiv_submodule` is the equivalence between `projectivization K V`
+  and `{ H : submodule K V // finrank H = 1 }`.
+- For `v : projectivization K V`, `v.rep : V` is a representative of `v`.
+
+-/
+
+variables (K V : Type*) [field K] [add_comm_group V] [module K V]
+
+/-- The setoid whose quotient is the projectivization of `V`. -/
+def projectivization_setoid : setoid { v : V // v ≠ 0 } :=
+{ r := λ u v, ∃ a : K, a • (u : V) = v,
+  iseqv := begin
+    refine ⟨λ u, ⟨1, by simp⟩, _, λ u v w ⟨a,ha⟩ ⟨b,hb⟩, ⟨b * a, by simp [ha, hb, mul_smul]⟩⟩,
+    rintros u v ⟨a,ha⟩,
+    use a⁻¹,
+    rw [← ha, ← mul_smul, inv_mul_cancel, one_smul],
+    intros c, apply v.2,
+    simpa [c] using ha.symm,
+  end }
+
+
+/-- The projectivization of the `K`-vector space `V`. -/
+def projectivization := quotient (projectivization_setoid K V)
```

Comment:
What do you think of calling this `projective_space` rather than `projectivization`?

Reply:
I prefer to use `projectivization` to emphasize that it's constructed out of a vector space.
There is a notion of `projective_space` with a more combinatorial definition, and various coordinatization results show that these "abstract" projective spaces can be identified with the projectivization of some vector space.

Reply:
OK, no objection.  I guess we can have notation `ℙ K V` (or maybe even `Kℙ V`, which would allow us to write `ℂℙ V` and `ℝℙ V`!!), so that the name doesn't really matter.